cdap-app-fabric/src/main/java/co/cask/cdap/gateway/handlers/NamespaceHttpHandler.java

```diff
@@ -145,12 +148,15 @@ public void create(HttpRequest request, HttpResponder responder, @PathParam("nam
   @DELETE
   @Path("/namespaces/{namespace-id}")
   public void delete(HttpRequest request, HttpResponder responder, @PathParam("namespace-id") String namespace) {
+    if (!conf.getBoolean(Constants.Dangerous.UNRECOVERABLE_RESET,
```

Comment:
This may only be temporary. Is ACLs/authorization for namespaces in the plan for 2.8? If so, @jwang47 thoughts on how to handle this?